.changeset/fuzzy-socks-return.md

```diff
@@ -0,0 +1,7 @@
+---
+"@itwin/itwinui-react": minor
+---
+
+* Clicking a `MenuItem` with a non empty `submenuItems` array now toggles the submenu visibility.
+  * Thus, passing a non empty `submenuItems` array and `onClick` to `MenuItem` at the same time is not supported.
+  * If both props are passed, clicking the `MenuItem` will toggle the submenu visibility but also *still* call the `onClick` (to prevent a behavioral change).
```

Comment:
"non-empty" is implied, and "supported" should be reworded as "recommended", and the warning should be mentioned.

also slight formatting nit around bullet point (since changesets will already add a bullet point).

```suggestion
Clicking a `MenuItem` with `submenuItems` now toggles the submenu visibility.
  * If both `submenuItems` and `onClick` props are passed, then clicking the `MenuItem` will toggle the submenu visibility but also *still* call the `onClick`. However, this behavior can lead to a confusing UX and is not recommended, so a warning will be shown.
```